tests/memoryview/memslice.pyx

```diff
@@ -1130,6 +1130,54 @@ def assign_temporary_to_object(object[:] buf):
     """
     buf[1] = {3-2: 2+(2*4)-2}
 
+@testcase
+def check_object_nulled_1d(object[:] buf, int idx, obj):
+    """
+    See comments on printbuf_object above.
+
+    >>> a = object()
+    >>> rc1 = get_refcount(a)
+    >>> A = ObjectMockBuffer(None, [a, a])
+    >>> check_object_nulled_1d(A, 0, a)
+    >>> decref(a)  # new reference "added" to A
+    >>> check_object_nulled_1d(A, 1, a)
+    >>> decref(a)
+    >>> A = ObjectMockBuffer(None, [a, a, a, a], strides=(2,))
+    >>> check_object_nulled_1d(A, 0, a)  # only 0 due to stride
+    >>> decref(a)
+    >>> get_refcount(a) == rc1
+    True
+    """
+    cdef ObjectMockBuffer omb = buf.base
+    cdef void **data = <void**>(omb.buffer)
+    data[idx] = NULL
```

Comment:
That's what the `deref` in the docstring is for I think. Although maybe it's a non-obvious place to put it

Reply:
I'd rather have the decref in a safe place. Otherwise, if something fails in the test, then the doctest would still try to apply the decref, which may not be correct and might trigger a hard crash, which is very annoying for remote debugging on CI.

Reply:
> doctest would still try to apply the decref, which may not be correct and might trigger a hard crash, which is very annoying for remote debugging on CI

True. It was a bit of a pain when debugging it locally... I'll move it when I next update this

Reply:
Done. Moved it to Py_CLEAR inside the function

Reply:
That's better. Thanks!